third_party/amd/lib/TritonAMDGPUTransforms/ConvertToBufferOps.cpp

```diff
@@ -323,6 +323,13 @@ struct ConvertTritonAtomicRMWOpToBufferAtomicRMW
     }
     LDBG("RMW supported type");
 
+    // float16 is the only 16-bit dtype supported by buffer atomic fadd
+    // TBD: Check if this should be skipped for gfx950
```

Comment:
Yes [gfx950 supports it](https://github.com/llvm/llvm-project/blob/24f432d33eb05175bd7237f9cac304afaba738a3/llvm/lib/Target/AMDGPU/AMDGPU.td#L1666). Can you guard this check for gfx942 and below?